### PR TITLE
Add Initial Qt6 Build Support

### PR DESCRIPTION
**Pull request by simpsont-oci**
Problem:
 - Build files are not currently set up to build for Qt6
 - A few changes in Qt6 APIs break the build for Qt6

Solution:
 - Expand build coverage to include Qt6
 - Fix build for Qt6 APIs

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -235,7 +241,9 @@ jobs:
           cd build
           CMAKE_ARGS=""
           if [ '${{ runner.os }}' == 'macOS' ]; then
-            CMAKE_ARGS+=" -DQt5_DIR=$(brew --prefix qt@5)/lib/cmake/Qt5"
+            # CMAKE_ARGS+=" -DQt5_DIR=$(brew --prefix qt@5)/lib/cmake/Qt5"
```
Comment (mitza-oci) [inline .github/workflows/build.yml:244]:
```suggestion
```

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.20)
 project(opendds-monitor-tests VERSION 0.0.1 LANGUAGES CXX)
 
 add_library(test_common INTERFACE)
-target_compile_options(test_common INTERFACE $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>: -Wall -Wpedantic -Wno-unused -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2 -Winit-self -Wmissing-include-dirs -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wstrict-overflow=5 -Wundef -Werror> $<$<CXX_COMPILER_ID:MSVC>: /W4>)
+target_compile_options(test_common INTERFACE $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>: -Wall -Wextra -Wpedantic -Wno-unused -Wunused-parameter> $<$<CXX_COMPILER_ID:MSVC>: /W4>)
```
Comment (mitza-oci) [inline test/CMakeLists.txt:6]: Is this required for Qt6 or were the additional switches redundant, or something else?

**Review by mitza-oci — CHANGES_REQUESTED** [review]

File: test/CMakeLists.txt (hunk 2 above)
Comment (simpsont-oci) [reply test/CMakeLists.txt:6]: The tests did fail to compile with these switches for Qt6, but they also didn't match the flags that are being used for the main monitor application, so I updated them to match and the problem (I think it was a strict overflow warning-become-error in a Qt header?) went away.

**Comment by simpsont-oci** [comment]
At some point, we should update the CI/CD workflows to actually verify this builds with Qt6. We can convert this to draft if you want me to look into that eventually. Or we can open an issue and someone can do it later.

**Pull request merged**